### DISCUSSION TITLE
What is dummy vk hash for?

### DISCUSSION THREAD

**Discussion by L-as** [MinaProtocol/mina#14438] [Q&A] [answered]
I'm looking at https://github.com/MinaProtocol/mina/blob/d5ae586319a9d0fdfd322b2b8b54beb5bfaeddbb/src/app/zkapps_examples/zkapps_examples.ml#L287

**Comment by nholland94** [comment]
It's just a placeholder for a verification key when you don't actually have one.

**Reply by L-as** [reply]
@nholland94 but it's used in-circuit, why? What happens if my public output for my zkapp circuit is a digest of such an account update?

**Reply by mrmr1993** [reply, marked as answer]
We need some value to insert into the circuit, even if you aren't using a proof, to make sure txns are homogeneous. This verification key doesn't admit any valid proofs, so attempting to send a proof with it will always fail; the goal is to allow signatures/no auth.

**Reply by L-as** [reply]
Unfortunate that it isn't a proper sum.
